spectral_cube/spectral_cube.py

```diff
@@ -1912,6 +1912,68 @@ def subcube_from_ds9region(self, ds9region, allow_empty=False):
         # that we can now crop out
         return masked_subcube.minimal_subcube(spatial_only=True)
 
+    def subcube_from_ds9region_new(self, ds9region, allow_empty=False):
+        """
+        Extract a masked subcube from a ds9 region or a Region list object
+        (only functions on celestial dimensions)
+
+        Parameters
+        ----------
+        ds9region: str or `regions.Region` list
+            The region to extract
+        allow_empty: bool
+            If this is False, an exception will be raised if the region
+            contains no overlap with the cube
+        """
+        import regions
+
+        if isinstance(ds9region, six.string_types):
+            shapelist = regions.DS9Parser(ds9region).shapes.to_regions()
+        else:
+            shapelist = ds9region
+
+        regionlist = []
+        for x in shapelist:
+            if isinstance(x, regions.SkyRegion):
+                regionlist.append(x.to_pixel(self.wcs.celestial))
+            else:
+                regionlist.append(x)
+
+        compound_region = _compound_region(regionlist)
+        mask = compound_region.to_mask()
+
+        xlo, xhi, ylo, yhi = mask.bbox.ixmin, mask.bbox.ixmax, mask.bbox.iymin, mask.bbox.iymax
+
+        # Negative indices will do bad things, like wrap around the cube
+        # If xhi/yhi are negative, there is not overlap
+        if (xhi < 0) or (yhi < 0):
+            raise ValueError("Region is outside of cube.")
+
+        if xlo < 0:
```

Comment:
Just so you're aware: the reason we have this check on the negative side is that indexing with negative numbers does weird and incorrect things.

Reply:
yes, it was mentioned in the comments there. Also since the pixel origin (0, 0) is at top left , it makes no sense to have negative pixels. Am I right??

Reply:
That's right.  Except, I regard 0,0 as bottom-left, which is the "image" standard and the one used in astronomy.  0,0 being top-left is the "matrix" standard, which is generally not used in our field.